sources/packages/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -79,6 +79,79 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
       .getOne();
   }
 
+  /**
+   * Get the assessment data to load the NOA (Notice of Assessment)
+   * for a student application.
+   * @param assessmentId assessment id to be retrieved.
+   * @param studentId associated student of the application. Provide
+   * when an authorization check is needed.
+   * @returns assessment NOA data.
+   */
+  async getAssessmentForNOA(
+    assessmentId: number,
+    studentId?: number,
+  ): Promise<StudentAssessment> {
+    const query = this.repo
+      .createQueryBuilder("assessment")
+      .select([
+        "assessment.assessmentData",
+        "application.applicationNumber",
+        "student.id",
+        "user.firstName",
+        "user.lastName",
+        "educationProgram.name",
+        "location.name",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "offering.offeringIntensity",
+        "msfaaNumber.msfaaNumber",
+        "disbursementSchedule.disbursementDate",
+        "disbursementValue.valueType",
+        "disbursementValue.valueCode",
+        "disbursementValue.valueAmount",
+      ])
+      .innerJoin("assessment.application", "application")
+      .innerJoin("application.student", "student")
+      .innerJoin("application.msfaaNumber", "msfaaNumber")
+      .innerJoin("student.user", "user")
+      .innerJoin("assessment.offering", "offering")
+      .innerJoin("offering.educationProgram", "educationProgram")
+      .innerJoin("offering.institutionLocation", "location")
+      .innerJoin("assessment.disbursementSchedules", "disbursementSchedule")
+      .innerJoin("disbursementSchedule.disbursementValues", "disbursementValue")
+      .innerJoin("disbursementSchedule.studentAssessment", "studentAssessment")
+      .where("assessment.id = :assessmentId", { assessmentId })
+      .orderBy("disbursementSchedule.disbursementDate");
+
+    if (studentId) {
+      query.andWhere("student.id = :studentId", { studentId });
+    }
+    return query.getOne();
+  }
+
+  /**
+   * Get the assessments associated with an application.
+   * @param applicationId application id.
+   * @param triggerType optional type to filter.
+   * @returns filtered assessments. Please note that 'Original assessment'
+   * will always return only one record.
+   */
+  async getAssessmentsByApplicationId(
+    applicationId: number,
+    triggerType?: AssessmentTriggerType,
+  ): Promise<StudentAssessment[]> {
+    const query = this.repo
+      .createQueryBuilder("assessment")
+      .select(["assessment.id"])
+      .innerJoin("assessment.application", "application")
+      .where("application.id = :applicationId", { applicationId });
+    if (triggerType) {
+      query.andWhere("assessment.triggerType = :triggerType", { triggerType });
+    }
+
```

Comment:
extra line

Reply:
Usually, we add some brack lines in the middle of the method body for the sake of making it clear. Here it is not the case that is why I am removing it, but in general it should not be a problem 😉